packages/itwinui-react/src/core/ComboBox/ComboBox.tsx

```diff
@@ -321,84 +343,58 @@ export const ComboBox = React.forwardRef(
             ),
         );
         if (focusedIndex != -1) {
-          dispatch({ type: 'focus', value: -1 });
+          setFocusedIndex(-1);
         }
         inputProps?.onChange?.(event);
       },
       [filterFunction, focusedIndex, inputProps, optionsRef, show],
     );
 
-    // When the value prop changes, update the selected index/indices
-    React.useEffect(() => {
```

Comment:
Where is all this code handled now?

Reply:
This is currently taken care of by the `useControlledState`. Whenever the `valueProp` changes, `useControlledState` updates the `value` accordingly.

Whenever the `value` changes, `selectedIndices` is updated because `value` is a part of the dependency array ([code](https://github.com/iTwin/iTwinUI/blob/6355ce0e8062070216e558d09a8cf361664c2a71/packages/itwinui-react/src/core/ComboBox/ComboBox.tsx#L234-L253)).